workers/unity/Packages/com.improbable.gdk.tools/GenerateCode.cs

```diff
@@ -219,75 +221,20 @@ private static void ForceGenerate()
-            try
+            // Get all packages we depend on
+            var request = Client.List(offlineMode: true);
+            while (!request.IsCompleted)
             {
-                var toolsConfig = GdkToolsConfiguration.GetOrCreateInstance();
-                // Safe as we validate there is at least one entry.
-                var schemaRoot = toolsConfig.SchemaSourceDirs[0];
-                CleanDestination(schemaRoot);
-
-                // Get all packages we depend on
-                var request = Client.List(offlineMode: true);
-                while (!request.IsCompleted)
-                {
-                    // Wait for the request to complete
-                }
-
-                var schemaSources = request.Result.ToDictionary(package => package.name,
-                        package => Path.Combine(package.resolvedPath, "Schema"))
-                    .Where(kv => Directory.Exists(kv.Value));
-
-                foreach (var source in schemaSources)
-                {
-                    CopySchemaFiles(schemaRoot, source.Key, source.Value);
-                }
+                // Wait for the request to complete
             }
-            catch (Exception e)
-            {
-                Console.Error.WriteLine(e.Message);
-                Environment.Exit(1);
-            }
-        }
-
-        private static void CopySchemaFiles(string schemaRoot, string packageName, string packageSchemaPath)
-        {
-            var files = Directory.GetFiles(packageSchemaPath, "*.schema", SearchOption.AllDirectories);
-            foreach (var file in files)
-            {
-                var relativeFilePath =
-                    file.Replace(packageSchemaPath, string.Empty).TrimStart(Path.DirectorySeparatorChar);
-                var fileCopy = Path.Combine(schemaRoot, FromGdkPackagesDir, packageName, relativeFilePath);
 
-                var directoryName = Path.GetDirectoryName(fileCopy);
-                if (!string.IsNullOrEmpty(directoryName))
-                {
-                    if (!Directory.Exists(directoryName))
-                    {
-                        Directory.CreateDirectory(directoryName);
-                    }
-                }
+            var packagePathsWithSchema = request.Result.Select(package => Path.Combine(package.resolvedPath, SchemaPackageDir)).Where(Directory.Exists);
 
-                using (var output = File.OpenWrite(fileCopy))
-                using (var outputText = new StreamWriter(output))
-                {
-                    outputText.Write(SchemaWarningMessage);
-                    using (var inputText = File.OpenRead(file))
-                    {
-                        inputText.CopyTo(outputText.BaseStream);
-                    }
-                }
-            }
-        }
+            var cachedPackagePathsWithSchema = Directory.GetDirectories("Library/PackageCache").Select(path => Path.GetFullPath(Path.Combine(path, SchemaPackageDir))).Where(Directory.Exists);
```

Comment:
Is this were the NPM packages are kept? Is it probably cleaned if you remove a package through the UI?

Reply:
Yes, and yes.
If you remove a package from the manifest, or change its version, it will be removed (and re-added).